group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -1350,6 +1351,11 @@ private CoordinatorResult<ConsumerGroupHeartbeatResponseData, Record> consumerGr
                         .withMembers(group.members())
                         .withStaticMembers(group.staticMembers())
                         .withSubscriptionMetadata(subscriptionMetadata)
+                        .withSubscriptionType(ConsumerGroup.subscriptionType(
+                            subscribedTopicsMemberCount,
+                            group.numMembers(),
```

Comment:
An alternative may be to compare the counts within the Map without considering the group size. This could work because we do not accept empty subscriptions from a member.

Reply:
yeah I was actually thinking the same, I saw that the member map is updated in the getOrCreateMember step so I figured the memberCount is updated, unless a member is removed, I'm not sure if that updates elsewhere, or I might be overlooking something

Reply:
An alternative may be to compare the counts within the Map without considering the group size. This could work because we do not accept empty subscriptions from a member. -> this works as well